app/pages/lab/visibility.cjsx

```diff
@@ -27,8 +31,22 @@ module.exports = React.createClass
   setterProperty: 'project'
 
   componentDidMount: ->
+    page = if @props.location?.query?.page? then @props.location.query.page else 1
+    @getWorkflowList page
+
+  onPageChange: (page) ->
+    nextQuery = Object.assign {}, @props.location.query, { page }
+    @context.router.push
+      pathname: @props.location.pathname
+      query: nextQuery
+    @getWorkflowList page
+    workflowList = document.getElementById("workflow-list")
```

Comment:
Use a [`ref`](https://facebook.github.io/react/docs/refs-and-the-dom.html) instead of directly interacting with the DOM.